x-pack/plugins/security_solution/server/lib/detection_engine/signals/filter_events_with_list.ts

```diff
@@ -51,84 +51,88 @@ export const filterEventsAgainstList = async ({
     );
 
     if (exceptionItemsWithLargeValueLists.length === 0) {
-      logger.debug(buildRuleMessage('about to return original search result'));
+      logger.debug(
+        buildRuleMessage('no exception items of type list found - returning original search result')
+      );
       return eventSearchResult;
     }
 
     // narrow unioned type to be single
     const isStringableType = (val: SearchTypes) =>
       ['string', 'number', 'boolean'].includes(typeof val);
-    // grab the signals with values found in the given exception lists.
-    const filteredHitsPromises = exceptionItemsWithLargeValueLists.map(
-      async (exceptionItem: ExceptionListItemSchema) => {
-        const { entries } = exceptionItem;
-
-        const filteredHitsEntries = entries
-          .filter((t): t is EntryList => entriesList.is(t))
-          .map(async (entry) => {
-            const { list, field, operator } = entry;
-            const { id, type } = list;
-
-            // acquire the list values we are checking for.
-            const valuesOfGivenType = eventSearchResult.hits.hits.reduce(
-              (acc, searchResultItem) => {
-                const valueField = get(field, searchResultItem._source);
-
-                if (valueField != null && isStringableType(valueField)) {
-                  acc.add(valueField.toString());
-                }
-                return acc;
-              },
-              new Set<string>()
-            );
 
-            // matched will contain any list items that matched with the
-            // values passed in from the Set.
-            const matchedListItems = await listClient.getListItemByValues({
-              listId: id,
-              type,
-              value: [...valuesOfGivenType],
-            });
-
-            // create a set of list values that were a hit - easier to work with
-            const matchedListItemsSet = new Set<SearchTypes>(
-              matchedListItems.map((item) => item.value)
-            );
+    // grab the signals with values found in the given exception lists.
+    const allExceptionsEntries = exceptionsList
+      .map((excItem: ExceptionListItemSchema) => {
+        const { entries } = excItem;
+        return entries;
+      })
+      .flat();
```

Comment:
We can't put all entries from all exception items in the same array here - we need to maintain the association between entries within an exception item.

The boolean logic this code is executing is essentially "filter out signals that match `(exc1_1 OR exc1_2 OR exc2_1 OR exc2_2 OR ...)`", where exc1_1 is the first list entry in exception item 1, exc1_2 is the second list entry, etc. We need to execute "filter out signals that match `(exc1_1 AND exc1_2) OR (exc2_1 AND exc2_2) OR ...`" - i.e. to be filtered out a signal has to match all entries within any single exception item.